src/app/components/navigation/home.rs

```diff
@@ -169,6 +228,27 @@ impl EventListener for HomePane {
             AppEvent::BrowserEvent(BrowserEvent::SavedPlaylistsUpdated) => {
                 self.update_playlists_in_sidebar();
             }
+            AppEvent::LoginEvent(LoginEvent::LoginStarted(event)) => {
```

Comment:
a lot of the complexity here (storing handlers, unbinding rebinding, etc) could be avoided I think! your model should be able to get the user id from the state instead of this, and then call spotify

Reply:
Very helpful, thanks. I just started discovering how the state worked a few days ago, that's the only reason why I didn't to it like that in the first place!